pkg/ebpf/derivations.go

```diff
@@ -0,0 +1,61 @@
+package ebpf
+
+import (
+	"github.com/aquasecurity/tracee/types/trace"
+)
+
+//returns the derived event, wether an event was derived, and a relevant error if failure was due to one
+type deriveFn func(trace.Event, EventDefinition) (trace.Event, bool, error)
+
+func deriveContainerCreate(t *Tracee) deriveFn {
```

Comment:
I see why we would want that but that can easily cause circular dependencies since most derivation function actually depend on it's tracee's object current state, so it's difficult to separate.
I'll see if it's possible since it would be better if we can do it.

Reply:
Actually, these aren't exported so there's not much of reason to be worried about usage outside of `ebpf`

Reply:
it is still more correct to have it under internal, you do not what anyways to have them used outside (no one should expose them)

Reply:
Resolved offline, this will be done in a later PR to keep it out of this one's scope.

Reply:
I would wait with putting it under "internal" of the ebpf package.
The ebpf package will eventually handle only ebpf related stuff - bpf maps population, program loading/attachment, bpf filters and perf buffer polling and decoding.
In the future, we can potentially split some of this logic to a new "events" package, and there we can have such "internal" packages